Scraping Windows service info crashes Alloy

### What's wrong?

Grafana Alloy works very smooth over all. Thanks for your work!

I ran into the problems descirbed in https://community.grafana.com/t/windows-exporter-component-crash-when-scrapping-scheduled-tasks/126616 so i upgraded to the latest rc version of Grafana Alloy. In general it works great. But i have problems scraping the windows service information. I couldn't figure out what exactly the problem is. But without enabling the "service" collector everything works fine. If you have any more questions surrounding the setup i'm happy to answer them. Grafana Alloy runs as Windows SYSTEM user while some services run under different Active Directory Accounts. Maybe the problem is related to some access restrictions?

### Steps to reproduce

See the configuration blocks below. Grafana Alloy crashes on the first windows integration scrape. So shortly after starting it (as a service / as a binary from cmd).

### System information

Windows Server 2022 Standard 21H2

### Software version

Grafana Alloy v1.4.0-rc.3

### Configuration

The configuration was migrated form the static Grafana Agent configuration.

This works:
```
prometheus.exporter.windows "integrations_windows_exporter" {
	enabled_collectors = ["cpu","cs","logical_disk","memory","net","os","process","scheduled_task","system"]

	process {
		whitelist = "Test.*"
	}
}
```

This does not work:
```
prometheus.exporter.windows "integrations_windows_exporter" {
	enabled_collectors = ["cpu","cs","logical_disk","memory","net","os","process","scheduled_task","service","system"]

	process {
		whitelist = "Test.*"
	}

	service {
		where_clause = "Name='TestService'"
	}
}
```

I also tried this. This works but it generates a warning (see below):
```
prometheus.exporter.windows "integrations_windows_exporter" {
	enabled_collectors = ["cpu","cs","logical_disk","memory","net","os","process","scheduled_task","service","system"]

	process {
		whitelist = "Test.*"
	}

	service {
		where_clause          = "Name='TestService'"
		enable_v2_collector   = "true"
		use_api               = "true"
	}
}
```

### Logs

Running Grafana Alloy with "service" collector enabled and only `where_clause` configured:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x7728d33]

goroutine 896 [running]:
github.com/prometheus-community/windows_exporter/pkg/collector/service.(*Collector).Collect(0xc001c11a90, 0x0?, 0x0?)
        /go/pkg/mod/github.com/prometheus-community/windows_exporter@v0.27.2/pkg/collector/service/service.go:152 +0x33
github.com/prometheus-community/windows_exporter/pkg/collector.(*Prometheus).execute(0xc00306f100, {0xa37368f, 0x7}, {0xba52528, 0xc001c11a90}, 0xc0000e6bb0, 0xc0049db980)
        /go/pkg/mod/github.com/prometheus-community/windows_exporter@v0.27.2/pkg/collector/prometheus.go:176 +0x8f
github.com/prometheus-community/windows_exporter/pkg/collector.(*Prometheus).Collect.func2({0xa37368f, 0x7}, {0xba52528?, 0xc001c11a90?})
        /go/pkg/mod/github.com/prometheus-community/windows_exporter@v0.27.2/pkg/collector/prometheus.go:117 +0xa5
created by github.com/prometheus-community/windows_exporter/pkg/collector.(*Prometheus).Collect in goroutine 880
        /go/pkg/mod/github.com/prometheus-community/windows_exporter@v0.27.2/pkg/collector/prometheus.go:115 +0x470
```
This works but generates a lot of warnings. Running Grafana Alloy with "service" collector enabled and all the properties configured:
```
ts=2024-09-25T08:30:14.9783748Z level=warn msg="API collection is enabled." component_path=/ component_id=prometheus.exporter.windows.integrations_windows_exporter collector=service
ts=2024-09-25T08:31:05.9236679Z level=warn msg="Get service McpManagementService config error:  0x3afc" component_path=/ component_id=prometheus.exporter.windows.integrations_windows_exporter collector=service
ts=2024-09-25T08:32:05.8807195Z level=warn msg="Get service McpManagementService config error:  0x3afc" component_path=/ component_id=prometheus.exporter.windows.integrations_windows_exporter collector=service
ts=2024-09-25T08:33:05.8822429Z level=warn msg="Get service McpManagementService config error:  0x3afc" component_path=/ component_id=prometheus.exporter.windows.integrations_windows_exporter collector=service
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,9 @@ Main (unreleased)
 
 ### Bugfixes
 
+- Update windows_exporter from v0.27.2 vo v0.27.3: (@jkroepke)
+  - Fixes a bug where scraping Windows service crashes alloy
+
 - Update yet-another-cloudwatch-exporter from v0.60.0 vo v0.61.0: (@morremeyer)
   - Fixes a bug where cloudwatch S3 metrics are reported as `0`
 

--- a/go.mod
+++ b/go.mod
@@ -149,7 +149,7 @@ require (
 	github.com/prometheus-community/elasticsearch_exporter v1.5.0
 	github.com/prometheus-community/postgres_exporter v0.11.1
 	github.com/prometheus-community/stackdriver_exporter v0.15.1
-	github.com/prometheus-community/windows_exporter v0.27.2
+	github.com/prometheus-community/windows_exporter v0.27.3
 	github.com/prometheus-operator/prometheus-operator v0.66.0
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.66.0
 	github.com/prometheus-operator/prometheus-operator/pkg/client v0.66.0
@@ -298,7 +298,7 @@ require (
 	github.com/Azure/go-autorest/logger v0.2.1 // indirect
 	github.com/Azure/go-autorest/tracing v0.6.0 // indirect
 	github.com/AzureAD/microsoft-authentication-library-for-go v1.2.2 // indirect
-	github.com/BurntSushi/toml v1.3.2 // indirect
+	github.com/BurntSushi/toml v1.3.2
 	github.com/ClickHouse/clickhouse-go v1.5.4 // indirect
 	github.com/Code-Hex/go-generics-cache v1.5.1 // indirect
 	github.com/DataDog/agent-payload/v5 v5.0.131 // indirect

